generics/setttl.go

```diff
@@ -0,0 +1,75 @@
+package generics
+
+import (
+	"cmp"
+	"sort"
+	"time"
+
+	"golang.org/x/exp/maps"
+)
+
+// SetWithTTL is a unique set of items with a TTL (time to live) for each item.
+// After the TTL expires, the item is automatically removed from the set.
+type SetWithTTL[T cmp.Ordered] struct {
```

Comment:
Should we make this implementation concurrent safe? If not, maybe we can add a note about it in the comments